### ISSUE TITLE
Error: NJS-009: invalid number of parameters

### ISSUE THREAD

**Issue by rstor1** [oracle/node-oracledb#998]
#### Answer the following questions:
**Note**: I did search other issues with same error and I noticed that one issue was because of a SELECT statement had a syntax error. The app I am working on has a few of these but they have not been altered in quite a while, so I am thinking it is something else. The error seems to propagate to or from where create pool is called with credentials in the error stack screenshot below. When I step through the code, it gets to a method `createPool(this.creds)` where `this.creds` is undefined at that point and errors out. I know it might sound trivial but I am not sure if this is the real reason because it does not crash the app.

1. What is your Node.js version?  Is it 64-bit or 32-bit?  **6.9.1 64-bit**

2. What is your node-oracledb version? **1.11.0**

3. What *exact* command caused the problem (e.g. what command did you try to install with)?  Who were you logged in as? **npm run start**

4. What error(s) you are seeing? **Error: NJS-009: invalid number of parameters**

5. What OS (and version) is Node.js executing on? **MacOS 10.13.6**

6. What is your Oracle client (e.g. Instant Client) version?  Is it 64-bit or 32-bit?  How was it installed?  Where is it installed? **12.2, installed in correct place as I had another question about this and got it cleared up.**

7. What is your Oracle Database version? **Not sure**

8. What is the `PATH` environment variable (on Windows) or `LD_LIBRARY_PATH` (on Linux) set to?  On macOS, what is in `~/lib`? 

**In lib:**

![in lib](https://user-images.githubusercontent.com/43070937/47218261-24ff1300-d379-11e8-88a9-9c04c0e0ed13.png)

9. What Oracle environment variables did you set?  How *exactly* did you set them?

10. Do you have a small, single Node.js script that immediately runs to show us the problem? **No but here is what the execute looks like:**

![invalid params](https://user-images.githubusercontent.com/43070937/47218287-3f38f100-d379-11e8-91cd-615998c7e4b3.png)

**Comment by dmcghan** [comment]
@rstor1 

> When I step through the code, it gets to a method createPool(this.creds) where this.creds is undefined at that point and errors out. 

Seems like you have an idea of what the problem is. Why are you calling create pool when `this.creds` doesn't have a value?

**Comment by rstor1** [comment]
@dmcghan this is a fairly complex app that I have been assigned to maintain/improve and to be honest I am not sure yet why it is being called without creds. I even tried setting the createPool at app startup and I still get errors like this after the fact. Gonna close this, was just making sure I was not missing something other than that. Thanks!

**Comment by rstor1** [comment]
Ok so after working on it some more I found that in order to debug my app locally, I had to use the embedded connection string technique suggested here: https://github.com/oracle/node-oracledb/issues/996
and adding an environment variable in the launch.json file in vscode helped too. No more errors. In debug mode it was not able to find the identifier specified for the remote oracledb oracledb because the environment variable was not set in debug mode. I think from here other errors were being triggered. So basically, setting an environment variable in launch.json and embedding the connection string for dev purposes did the trick.
